core/templates/pages/exploration-editor-page/translation-tab/state-translation/state-translation.component.ts

```diff
@@ -189,12 +205,19 @@ angular.module('oppia').component('stateTranslation', {
           );
           activeContentId = activeContent.contentId;
           if (activeContent instanceof SubtitledUnicode) {
-            activeDataFormat = WRITTEN_TRANSLATION_TYPE_UNICODE;
+            activeDataFormat = TRANSLATION_DATA_FORMAT_UNICODE;
           }
+        } else if (tabId === $scope.TAB_ID_RULE_INPUTS) {
+          const {
+            rule, inputName, contentId
+          } = $scope.interactionRuleTranslatableContents[0];
```

Comment:
Would this throw an error if no rules were defined or would it take default outcome rule's values (I'm assuming 0 means the rule index)?

Reply:
The structure of $scope.interactionRuleTranslatableContents is defined by getInteractionRuleTranslatableContents() in this file. It contains a list of {rule/inputname/contentid} that have translatable input types.

So yup, this would error in that case. However, the Rules tab is already grayed out anyway in such cases, so the click on the tab never happens.

Given that, I'm not entirely sure what the best thing to do is here. Do you think I should still add an exception check?

Reply:
Hmm, maybe add it anyway? Accessing empty arrays should throw an error directly, but I think it may lead to security issues also.

Reply:
Can we not render the tab if it is not clickable / grayed out?

Reply:
Hm, I think it's fine to keep it grayed out. That is the convention that is already present in the translation tab, so let's stick with it.

@aks681 I added a check, thanks!